### DISCUSSION TITLE
What can I do to ensure complete Sentry data scrubbing in Django?

### DISCUSSION THREAD

**Discussion by tal-zvon** [getsentry/support#4] [Support] [answered]
I built a Django app for a hospital. The app will store confidential patient records that I should never, ever see.
I won't have access to the production server - the hospital's IT staff will set it up, and run it.

I need some way to be notified about any issues the app runs into without any chance of exposing any patient records to me.

Sentry is definitely great for reporting problems a Django app might run into, and it has dedicated pages covering the different ways to do data scrubbing, which is awesome, but I need to be sure I haven't missed anything.

I basically need for Sentry to tell me which part of my code broke, without telling me anything else. Not anything coming from the database, no local variables, or mysql queries, or anything.

My configuration in my Django `settings.py` looks like this right now:

```python
sentry_sdk.init(
    dsn = "...",
    integrations=[DjangoIntegration()],
    traces_sample_rate=1.0,
    send_default_pii=False,
    with_locals=False,
    request_bodies="never",
    default_integrations=False,
)
```

I'm not using the `before_send` hook because from what I understand, I need to know what I'm looking for in order for the function being called by the hook to remove it from the data being sent to the sentry server. I have no idea what I would be looking for. Just because one test log entry being sent to Sentry doesn't have confidential data attached doesn't mean that a real error in a completely different part of the app won't have confidential data attached. Or am I wrong?

I have a few related questions:

1. How do I disable breadcrumbs?
   - It seems like there's a chance they might leak some private patient data
2. Do I need the `before_send` hook or anything? Or can I get away with just using the above settings?
3. Is there a way for the Sentry logs to be sent to a sentry.io account that is controlled by the hospital's IT staff, so that they may review a log, and then somehow forward that to me once they are satisfied it has no patient data attached?
4. Are my settings here enough to be reasonably sure that no patient data will end up on the Sentry server?

Thanks

Update: It looks like I can use `before_send` hook to delete breadcrumbs. Is that the preferred/only way to do it? There's no arg for the `sentry_sdk.init` to disable breadcrumbs?

**Comment by markstory** [comment, marked as answer]
> How do I disable breadcrumbs? 

Using the Django integration will instrument the django ORM to capture SQL queries. It sounds like that is something you don't want. Given that you want as little information as possible you should probably forgo using integrations which collect additional information about your application.

As for disabling breadcrumbs entirely, the `before_send` hook is the most effective way to remove all breadcrumbs. There is also a `before_breadcrumb` option that takes a function. The provided function allows you to filter the content of breadcrumbs as they are collected.

> Is there a way for the Sentry logs to be sent to a sentry.io account that is controlled by the hospital's IT staff, so that they may review a log, and then somehow forward that to me once they are satisfied it has no patient data attached?

That isn't a flow we support. You could run a self-hosted sentry though. That would let you run sentry in an environment your team controls more fully.

> Are my settings here enough to be reasonably sure that no patient data will end up on the Sentry server?

There isn't a way anyone but you can answer that question with any confidence. Your application could have sensitive data in error message strings, or log messages that we have no visibility into.
